components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -416,6 +416,29 @@ public String addIcon(String resourcePath, Icon icon) throws APIManagementExcept
         return null;
     }
 
+    /**
+     * Checks whether the given document already exists for the given api
+     *
+     * @param identifier API Identifier
+     * @param docName Name of the document
+     * @return true if document already exists for the given api
+     * @throws APIManagementException if failed to check existence of the documentation
+     */
+    public boolean isDocumentationExist(APIIdentifier identifier, String docName)
+            throws APIManagementException {
+        String docPath = APIConstants.API_ROOT_LOCATION + RegistryConstants.PATH_SEPARATOR +
+                identifier.getProviderName() + RegistryConstants.PATH_SEPARATOR +
+                identifier.getApiName() + RegistryConstants.PATH_SEPARATOR +
+                identifier.getVersion() + RegistryConstants.PATH_SEPARATOR +
+                APIConstants.DOC_DIR + RegistryConstants.PATH_SEPARATOR + docName;
+        try {
+            return registry.resourceExists(docPath);
+        } catch (RegistryException e) {
+            handleException("Failed to check existence of the document :" + docPath, e);
+            return false;
```

Comment:
handleException method throws an APIManagementException. Why do we have a return here?